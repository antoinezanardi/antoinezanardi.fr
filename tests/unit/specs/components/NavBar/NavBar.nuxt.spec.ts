import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { NavBar } from "#components";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

describe("NavBar Component", () => {
  let wrapper: ReturnType<typeof mount<typeof NavBar>>;

  async function mountNavBarComponent(options: ComponentMountingOptions<typeof NavBar> = {}): Promise<ReturnType<typeof mount<typeof NavBar>>> {
    return mountSuspendedComponent(NavBar, options);
  }

  beforeEach(async() => {
    wrapper = await mountNavBarComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });
});