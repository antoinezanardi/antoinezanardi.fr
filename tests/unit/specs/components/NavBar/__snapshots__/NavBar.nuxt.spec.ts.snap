// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Component > should match snapshot when rendered. 1`] = `
"<header data-v-fe2af0c4="">
  <div data-v-fe2af0c4="" class="profile-page sidebar-collapse">
    <nav data-v-fe2af0c4="" class="bg-primary fixed-top navbar navbar-expand-lg" data-color-on-scroll="100">
      <div data-v-fe2af0c4="" class="container">
        <div data-v-fe2af0c4="" class="navbar-translate"><a data-v-fe2af0c4="" class="navbar-brand text-uppercase" href="#">Antoine Zanardi</a><button data-v-fe2af0c4="" aria-controls="navigation" aria-expanded="false" aria-label="Ouvrir la navigation" class="navbar-toggler" data-target="#navigation" data-toggle="collapse" type="button"><span data-v-fe2af0c4="" class="bar1 navbar-toggler-bar"></span><span data-v-fe2af0c4="" class="bar2 navbar-toggler-bar"></span><span data-v-fe2af0c4="" class="bar3 navbar-toggler-bar"></span></button></div>
        <div data-v-fe2af0c4="" id="navigation" class="collapse justify-content-end navbar-collapse">
          <ul data-v-fe2af0c4="" class="navbar-nav w-100">
            <li data-v-fe2af0c4="" class="nav-item"><a data-v-fe2af0c4="" class="nav-link smooth-scroll" href="#about">Qui suis-je ?</a></li>
            <li data-v-fe2af0c4="" class="nav-item"><a data-v-fe2af0c4="" class="nav-link smooth-scroll" href="#skills">Compétences techniques</a></li>
            <li data-v-fe2af0c4="" class="nav-item"><a data-v-fe2af0c4="" class="nav-link smooth-scroll" href="#portfolio">Portfolio</a></li>
            <li data-v-fe2af0c4="" class="nav-item"><a data-v-fe2af0c4="" class="nav-link smooth-scroll" href="#experience">Expérience professionnelle</a></li>
            <li data-v-fe2af0c4="" class="nav-item"><a data-v-fe2af0c4="" class="nav-link smooth-scroll" href="#education">Formation &amp; Certifications</a></li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
</header>"
`;
